src/invidious.cr

```diff
@@ -2701,7 +2701,7 @@ get "/feed/private" do |env|
 
   env.response.content_type = "application/atom+xml"
 
-  token = env.params.query["token"]?
+  token = env.params.query.delete("token")
```

Comment:
You should avoid modifying `env.params.query` since it's used elsewhere (namely for logging). This change for example turns logs from:

```
200 GET /feed/private?token=XXXXX
```
to
```
200 GET /feed/private
```

Reply:
Oh, I did not know that! Maybe we can call this a *happy accident*: In my opinion, user tokens should not be logged. 

If that's not acceptable, I'll copy env.params.query before modifying it.

Reply:
The reason I would like this feature to use a separate parameter is because it removes any need to handle unsupported values (`token`, `page`...) by modifying `env.params.query`.

For example, something like:

```
/feed/private?token=a&params=speed%3D4%26local%3Dtrue
```

Could be consumed like:

```crystal
params = HTTP::Params.parse(env.params.query["params"]? || "")
video.to_xml(locale, host_url, params, xml)
```

Reply:
good arguments. implemented!